[CONS-6524] Standardize OpenShift examples (#1374)
DataDog/datadog-operator: docs/install-openshift.md

2. Create a file with the manifest of your `DatadogAgent` deployment.

> The following is the simplest recommended configuration for the `DatadogAgent` in OpenShift. Datadog recommends to set your `global.clusterName` accordingly:

Reviewer: ```suggestion
The following is the simplest recommended configuration for the `DatadogAgent` in OpenShift. Datadog recommends setting your `global.clusterName` as follows:
```

  ```yaml
  kind: DatadogAgent
  apiVersion: datadoghq.com/v2alpha1
  metadata:
    name: datadog

>     namespace: openshift-operators # set as namespace as where the Datadog Operator was deployed

Reviewer: ```suggestion
    namespace: openshift-operators # set as the same namespace where the Datadog Operator was deployed
```

        appSecret:
          keyName: app-key
          secretName: datadog-secret
      clusterName: <CLUSTER_NAME>
      kubelet:
        tlsVerify: false
    override:
      clusterAgent:
        serviceAccountName: datadog-agent-scc
      nodeAgent:
        serviceAccountName: datadog-agent-scc
        hostNetwork: true
        tolerations:
          - key: node-role.kubernetes.io/master
            operator: Exists
            effect: NoSchedule
          - key: node-role.kubernetes.io/infra
            operator: Exists
            effect: NoSchedule
  ```

> A few notes on this configuration:

Reviewer: ```suggestion
A few notes on this configuration:

```

- Setting the `serviceAccountName: datadog-agent-scc` in the `nodeAgent` and `clusterAgent` `override` section ensures that these pods are associated with the necessary `SecurityContextConstraints` and RBACs managed by the OperatorHub installation.

> - The Kubelet API certificates may not always be signed by cluster CA, so Kubelet TLS verification must be disabled.

Reviewer: ```suggestion
- The Kubelet API certificates might not always be signed by cluster CA. As a result, you must disable Kubelet TLS verification.
```

- In OpenShift 4.0+, set the `hostNetwork` parameter to allow access to your cloud provider metadata (IMDS) API endpoint for host tags and aliases.

> - If using APM or DogStatsD, disable the Unix Domain Socket (UDS) method as this requires highly elevated privileges on your APM application pods. Disabling it will also ensure the Admission Controller will not inject this configuration. To disable APM entirely set `features.apm.enabled` to false.

Reviewer: ```suggestion
- If using APM or DogStatsD, disable the Unix Domain Socket (UDS) method, as this requires highly elevated privileges on your APM application pods. Disabling it also ensures that the Admission Controller does not inject this configuration. To disable APM entirely, set `features.apm.enabled` to false.
```
